rclpy/src/rclpy/event.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <pybind11/pybind11.h>
+
+#include <rcl/error_handling.h>
+#include <rcl/rcl.h>
+#include <rcl/time.h>
+#include <rcl/types.h>
+
+#include <condition_variable>
+#include <cstring>
+#include <memory>
+#include <stdexcept>
+
+#include "event.hpp"
+
+namespace py = pybind11;
+
+namespace rclpy
+{
+template<typename ClockType>
+void Event::wait_until(std::shared_ptr<Clock> clock, rcl_time_point_t until)
+{
+  // Synchronize because clock epochs might differ
+  const rcl_time_point_t rcl_entry = clock->get_now();
+  const typename ClockType::time_point chrono_entry = ClockType::now();
+
+  rcl_duration_t delta_t;
+  rcl_ret_t ret = rcl_difference_times(&rcl_entry, &until, &delta_t);
+
+  if (RCL_RET_OK != ret) {
+    throw RCLError("failed to subtract times");
+  }
+
+  // Cast because system clock resolution is be too big for nanoseconds on Windows & OSX
```

Comment:
```suggestion
  // Cast because system clock resolution is too big for nanoseconds on Windows & OSX
```

Reply:
Removed 'be' in 9c93f3433ba76241c9c3b1932789d0adbd47e6a7